pkg/config/config.go

```diff
@@ -867,6 +867,11 @@ func loadProxyFromEnv(config Config) {
 		config.Set("proxy.no_proxy", p.NoProxy)
 		proxies = p
 	}
+
+	if !config.GetBool("proxy.use_proxy_for_host_metadata") {
```

Comment:
Where is this defined in the config object ? We need to set a default and bind it to the environment.